src/transformers/models/musicgen/modeling_musicgen.py

```diff
@@ -2533,6 +2550,16 @@ def resize_token_embeddings(self, *args, **kwargs):
             " model.decoder.resize_token_embeddings(...))"
         )
 
+    def freeze_encoders(self, freeze_text_encoder=True):
+        if freeze_text_encoder:
+            for param in self.text_encoder.parameters():
+                param.requires_grad = False
+            self.text_encoder._requires_grad = False
+
+        for param in self.audio_encoder.parameters():
+            param.requires_grad = False
+        self.audio_encoder._requires_grad = False
```

Comment:
I'm not a fan of this structure. In general, we don't add `freeze` methods to our models add leave that to the user to handle - although I see audio models appear to be the exception! 

It's tidier to split this up to `freeze_text_encoder` and `freeze_audio_encoder` and then just call them separately or add an additional `freeze_audio_encoder` argument

Reply:
I've split these up !